internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -119,7 +146,33 @@ func (g *Generator) Generate(opts ...Option) (err error) {
 
 	// Add extra annotations to csv
 	g.setAnnotations(csv)
-
+	// If a reader is set, and there is a flag to not update createdAt, then
+	// set the CSV's createdAt to the previous CSV's createdAt if its the only change.
+	if g.ignoreIfOnlyCreatedAtChanged && g.getReader != nil {
+		r, err := g.getReader()
+		if err != nil {
+			return err
+		}
+		var prevCSV v1alpha1.ClusterServiceVersion
+		err = genutil.ReadObject(r, &prevCSV)
+		if err != nil {
+			return err
+		}
+		if prevCSV.ObjectMeta.Annotations != nil && prevCSV.ObjectMeta.Annotations["createdAt"] != "" {
+			csvWithoutCreatedAtChange := csv.DeepCopy()
+			// Set WebhookDefinitions if nil to avoid diffing on it
+			if prevCSV.Spec.WebhookDefinitions == nil {
+				prevCSV.Spec.WebhookDefinitions = []v1alpha1.WebhookDescription{}
+			}
+			if csvWithoutCreatedAtChange.ObjectMeta.Annotations == nil {
+				csvWithoutCreatedAtChange.ObjectMeta.Annotations = map[string]string{}
+			}
+			csvWithoutCreatedAtChange.ObjectMeta.Annotations["createdAt"] = prevCSV.ObjectMeta.Annotations["createdAt"]
```

Comment:
We do have the code for walking through the input directories, collecting the manifests and unmarshalling it into the CSV in [here](https://github.com/operator-framework/operator-sdk/blob/78c564319585c0c348d1d7d9bbfeed1098fab006/internal/generate/collector/manifests.go#L192-L201). We should be able to fetch the createdAt at that step from annotations, and accordingly set it in [here](https://github.com/operator-framework/operator-sdk/blob/78c564319585c0c348d1d7d9bbfeed1098fab006/internal/generate/clusterserviceversion/clusterserviceversion_updaters.go#L67). This would avoid us doing the comparisons with webhooks done below.

Another option is, once we identify the CSV manifest, we can unmarshal it into a `runtime.Object` as done in this PR, and pass it on through the collector, to apply it in [here](https://github.com/operator-framework/operator-sdk/blob/78c564319585c0c348d1d7d9bbfeed1098fab006/internal/generate/clusterserviceversion/clusterserviceversion_updaters.go#L55). 

That would make sure that we are performing all operations related to inputs being passed on through flags in place and not read the input CSV multiple times. 

Wdyt?

Reply:
That sounds more efficient. Will have a look. Thanks!